### ISSUE TITLE
Can't add anything to dashboard.

### ISSUE THREAD

**Issue by asperkins** [capitalone/Hygieia#52]
Installed Mongo
Downloaded Hygieia V1.2.0
Successfully ran mvn clean compile install package
Created application.properties in corresponding config folders for Hudson Collector and Subversion Collector
Run both hudson and subversion collector with java -jar
Run gulp serve in the UI folder

Everything appears to come up ok
I see data has been created in mongo
I can add a dashboard in the UI

But, I can not succesffuly add any widgets via the UI. 

For subversion the dialog box comes up and and can select subversion and provide a url, but it will not save.
Debugging the JS I see

		function submitForm(valid, url, repoType) {
			ctrl.submitted = true;
			if (valid && ctrl.collectors.length) {

ctrl.collectors.length is 0

For the build widget, it comes up but there are no options avaible in the "Select a build job" drop down.

I've reviewed the documentation several time, but I feel like I'm missing some simple step.

Thanks,

Tony Perkins
Fidelity Investments

![image](https://cloud.githubusercontent.com/assets/4908783/9201567/5732116c-4013-11e5-82fe-4a7f5e5f4ce1.png)

![image](https://cloud.githubusercontent.com/assets/4908783/9201606/7d9c5a6a-4013-11e5-981b-815a644c2c6c.png)

**Comment by amitmawkin** [comment]
Tony,
Great progress looks like you are almost there  , can you make sure the collectors are running against your Hudson/Jenkins instance and same for Subversion, the collectors help populate the drop downs in the screen shot you have sent .

**Comment by asperkins** [comment]
The collector seems to be running ok. If found the jobs initially. 
![image](https://cloud.githubusercontent.com/assets/4908783/9225257/019dec7e-40ce-11e5-9c82-0168f48025a3.png)

I see data in mongo.

![image](https://cloud.githubusercontent.com/assets/4908783/9225262/0b00672e-40ce-11e5-98c6-94fe0b958062.png)

![image](https://cloud.githubusercontent.com/assets/4908783/9225281/27a9f3ea-40ce-11e5-98ea-d185aa592bae.png)

Not sure why it says "enabled":false tho

**Comment by tabladrum** [comment]
You can now select the build job in the Build Widget config. This will turn the HudsonCollectorItem's enabled (your screen shot above) to true. 

The idea is this: Gather all the build job names and URLs so that the drop down can be populated- but don't collect build data yet. Collect build data only for the jobs that have been configured on dashboards.

Let us know how it goes.

**Comment by asperkins** [comment]
This is where the problem is. Even though I see the collector running and the data in mongo, but the drop down in the widget still does not populate.

**Comment by tabladrum** [comment]
Do you get a json back from this: localhost:8080/api/dashboard?

**Comment by asperkins** [comment]
![image](https://cloud.githubusercontent.com/assets/4908783/9227182/dd0bf266-40d7-11e5-961c-fe9eb6a6010b.png)

**Comment by mgijsbertihodenpijl** [comment]
Yep

[{"id":"55cb3dab5b6660b9e4bedf21","template":"capone","title":"My first dashboard","application":{"name":"Jumbo","components":[{"id":"55cb3dab5b6660b9e4bedf20","name":"Jumbo","collectorItems":{}}]},"widgets":[],"owner":"admin"}]

**Comment by tabladrum** [comment]
I am on gitter now - join me on gitter - (badge right below the Hygieia logo on GH)

**Comment by mgijsbertihodenpijl** [comment]
mmm I think I found something ...

In API mongoconfig the mongo credentials are hardcoded:
        databaseName = properties.getProperty("dbname", "dashboard");
        host = properties.getProperty("dbhost", "localhost");
        port = Integer.parseInt( properties.getProperty("dbport", "27017") );
        userName = properties.getProperty("dbusername", "");
        password = properties.getProperty("dbpassword", "").toCharArray();
        System.out.println("dbusername=" + userName + ", password=" + password.toString());

I used 'dashboarddb' as database as I followed the api REAME example [dashboarddb].

**Comment by tabladrum** [comment]
@asperkins reported he had two databases in Mongo: dashboard and dashboarddb.
He removed the dashboard database things worked. I will check what created the dashboard database.

**Comment by asperkins** [comment]
Actually I changed my collectors to use dashboard instead of dashboarddb. I think the api component may not be reading the properties file.

**Comment by mgijsbertihodenpijl** [comment]
I got the Github collector and widget working.

The api component uses hard-coded properties in the  com.capitalone.dashboard.config.MongoConfig. So, I changed application.properties of the Github collector so it uses the same database ('dashboard') as MongoConfig. During startup, the collector is registered in the dashboard collection. Then I could add a Github widget in the UI. 

It would help if there is a simple step-by-step tutorial which shows all the steps needed from building to create a new widget. The dashboard.properties example  '[dashboarddb]' in the README in API module put me on the wrong foot.

**Comment by jdamick** [comment]
agreed - we need a better getting started guide

**Comment by golokeshpatra45** [comment]
I am facing problem with my Github collector and the data it should show on the dashboard.

Please see it once - 
![image](https://cloud.githubusercontent.com/assets/26374269/24655150/aa443804-195a-11e7-8b0f-04f4f398f050.png)

Now  I have tried to add repo link to get a representation on the Dashboard UI, but I am not able -
![image](https://cloud.githubusercontent.com/assets/26374269/24655105/66025630-195a-11e7-8ef2-896cdfecf395.png)

In addition to that, The drop down boxes are empty ,, help me out please
![image](https://cloud.githubusercontent.com/assets/26374269/24655276/3b6a050c-195b-11e7-87d7-0918ff1cb3eb.png)

![image](https://cloud.githubusercontent.com/assets/26374269/24655298/580cb272-195b-11e7-8a6f-f38f7d301ba8.png)

How to get this github collector working ?

**Comment by admiresail** [comment]
Unable to configure my github collector: 
When I tried to enter my URL and click "SAVE", nothing happens. 
I changed my db to "dashboard" and used the same in the collector.properties file and tried again but didn't work.  This is my api contents. Please help with your inputs in configuring the same.
![capture3](https://user-images.githubusercontent.com/33215272/50626173-bcc0d600-0f52-11e9-997a-1f5dd7099045.PNG)

**Comment by admiresail** [comment]
Hurray.. I was able to configure the GIT dashboard.  This is what I did:

1. Switched to "dashboard" db.
2. Created the user in "dashboard" db.
3. Changed my application.properties file and dashboard.properties file to point to "dashboard" db.
4. Ran my API , gulp serve, and my collector again.
5. Deleted my cookies and browsing history and temp files and logged into my dashboard.
6. Configured the scm widget and it worked. 

The tutorial should be updated with proper steps. I was going round and round with API, collector and DB for the past 2 days. Thank you all for your help !!! :) :) :) 👍
